assets/pages/new-contributions-landing/components/ButtonWithRightArrow.jsx

```diff
@@ -17,20 +17,21 @@ type PropTypes = {
 // ----- Render ----- //
 
 function ButtonWithRightArrow(props: PropTypes) {
+  const buttonElem = (
+    <button
+      className={props.buttonClassName}
+      type={props.type}
+      aria-describedby={props.accessibilityHintId}
+      onClick={props.onClick}
+    >
+      {props.buttonCopy}
+      <p id={props.accessibilityHintId} className="accessibility-hint">{props.buttonCopy}</p>
+      <SvgArrowRight />
+    </button>);
 
   return (
     <div className={props.componentClassName}>
-      <button
-        className={props.buttonClassName}
-        type={props.type}
-        href={props.url}
-        aria-describedby={props.accessibilityHintId}
-        onClick={props.onClick}
-      >
-        {props.buttonCopy}
-        <p id={props.accessibilityHintId} className="accessibility-hint">{props.buttonCopy}</p>
-        <SvgArrowRight />
-      </button>
+      { props.url ? <a href={props.url}>{buttonElem}</a> : buttonElem }
```

Comment:
Putting a `<button>` inside an `<a>` isn't actually valid HTML5 so it's behaviour may be unpredictable across browsers (http://w3c.github.io/html/textlevel-semantics.html#the-a-element says there must be no [interactive content](http://w3c.github.io/html/dom.html#interactive-content-2) descendants, which includes `<button>`)

Reply:
Not to mention the fact that you could pass both an `onClick` to the button and an `href` for the link - not sure what the behaviour would be then!

Reply:
This component feels a bit overloaded. I think the best option is to create a separate `LinkWithRightArrow` component which takes `url` but not `onClick`. Hopefully you can just apply the class [`button`](https://github.com/guardian/support-frontend/blob/master/assets/pages/new-contributions-landing/contributionsLanding.scss#L927-L937) to it to get the styling, or if not at least re-use most of what's in there

Reply:
There is a `ctaLink` that will do nicely for the Thank You page use case. Can then remove the `props.url` .